pages/sign-up.js

```diff
@@ -29,26 +48,13 @@ const SignUp = () => {
                         </p>
                       </div>
                       <div className="">
-                        <form action="#" method="POST" className="space-y-1">
-                          <div>
-                            <Input label="Email address" name="email" type="email" className="w-full" />
-                          </div>
-                          <div className="">
-                            <Input
-                              label="Password"
-                              name="password"
-                              type="password"
-                              Icon={<PasswordCheckIcon width="18px" height="18px" fill="#292D32" />}
-                            />
-                          </div>
-                          <div className="">
-                            <Input
-                              label="Confirm Password"
-                              name="confirm-password"
-                              type="password"
-                              Icon={<PasswordCheckIcon width="18px" height="18px" fill="#292D32" />}
-                            />
-                          </div>
+                        <HookForm onSubmit={onSubmit} schema={SignUpSchema} className="space-y-1">
+                          <Input label="Email address" name="email" type="email" />
+
+                          <Input label="Password" name="password" type="password" />
+
+                          <Input label="Confirm Password" name="confirmPassword" type="password" />
+
                           <div className="">
                             <div className="flex items-center mt-8 mb-3">
                               <input
```

Comment:
Can we create a custom react hook form component for Checkbox and Radio buttons too? Can you give this a trial?